src/dbnode/generated/thrift/rpc.thrift

```diff
@@ -29,12 +29,19 @@ enum TimeType {
 
 enum ErrorType {
 	INTERNAL_ERROR,
-	BAD_REQUEST
+	BAD_REQUEST,
+	RESOURCE_EXHAUSTED
+}
+
+enum ErrorCode {
+    NONE,
+    RESOURCE_EXHAUSTED
 }
```

Comment:
Some thoughts on backwards compatibility:

`ErrorType.RESOURCE_EXHAUSTED` cannot be used immediately, because older clients would fallback on `INTERNAL_ERROR`, which most likely would result in 500 responses. Having in mind that this would happen during cluster upgrade, it might be mistaken for faulty upgrade.
 
Similar regression might happen when changing error codes. E.g., if we wanted to make the error code more specific and introduced `QUERY_LIMIT_EXCEEDED` in place of `RESOURCE_EXHAUSTED`, during upgrade, older clients would be falling-back to `ErrorType` (assuming that the client code is written in appropriate way) until they are upgraded. It might be possible to avoid that by introducing binary coding for error codes, e.g.:

```
enum ErrorCode {
    NONE = 0x00,
    RESOURCE_EXHAUSTED = 0x01,
    QUERY_LIMIT_EXCEEDED = 0x0101,
    SOMETHING_ELSE_EXCEEDED = 0x0201
}
```

```go
func IsResourceExhaustedErrorCode(err rpc.Error) bool {
    return err != nil && (err.Code & RESOURCE_EXHAUSTED) == RESOURCE_EXHAUSTED 
}
```

Though I find it hard to decide if it wouldn't be an overkill.